### PR TITLE
[Tests] Fix smoke test for GCP disk tier `best`

### LINKED ISSUE #3973: [GCP] `test_gcp_disk_tier` for `best` tier failed due to best tier not mapping to pd-extreme
This issue is introduced by #3933. cc'ing @cblmemo 

_Version & Commit info:_
* `sky -v`: PLEASE_FILL_IN
* `sky -c`: PLEASE_FILL_IN

### PR DESCRIPTION
**Pull request by cblmemo**
Fixes #3973.

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [x] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_gcp_disk_tier` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: tests/test_smoke.py [hunk 1]
```diff
@@ -3374,11 +3374,15 @@ def test_gcp_disk_tier():
         name_on_cloud = common_utils.make_cluster_name_on_cloud(
             name, sky.GCP.max_cluster_name_length())
         region = 'us-west2'
+        instance_type_option = ''
+        if disk_tier == resources_utils.DiskTier.BEST:
+            # Ultra disk tier requires a specific instance type.
```
Comment (Michaelvll) [inline tests/test_smoke.py:3379]: Please confirm:
```suggestion
            # Ultra disk tier requires n2 instance types to have more than 64 CPUs.
```

Comment (Michaelvll) [reply tests/test_smoke.py:3379]: Also, could we also test that with `BEST` and the instance type not supporting it, i.e. we are using the correct `high` tier for those instance types?

**Review by Michaelvll — COMMENTED** [review]
Thanks for the fix @cblmemo!

File: tests/test_smoke.py (hunk 1 above)
Comment (cblmemo) [reply tests/test_smoke.py]: Done! And the smoke test passed as well.

File: tests/test_smoke.py [hunk 2]
```diff
@@ -3369,26 +3369,37 @@ def _get_aws_query_command(region, instance_id, field, expected):
...
-        run_one_test(test)
+        instance_type_options = ['']
+        if disk_tier == resources_utils.DiskTier.BEST:
+            # Ultra disk tier requires n2 instance types to have more than 64 CPUs.
+            # If using default instance type, it will only enable the high disk tier.
+            disk_types = [
+                GCP._get_disk_type(resources_utils.DiskTier.HIGH),
+                GCP._get_disk_type(resources_utils.DiskTier.ULTRA),
+            ]
+            instance_type_options.append('--instance-type n2-standard-64')
```
Comment (Michaelvll) [inline tests/test_smoke.py:3385]: Easier to read:
```suggestion
            instance_type_options = ['', '--instance-type n2-standard-64']
```

**Review by Michaelvll — APPROVED** [review]
Thanks @cblmemo! LGTM

**Pull request merged**